hazelcast/src/test/java/com/hazelcast/map/mapstore/MapLoaderFailoverTest.java

```diff
@@ -0,0 +1,197 @@
+/*
+ * Copyright (c) 2008-2015, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.map.mapstore;
+
+
+import com.hazelcast.config.Config;
+import com.hazelcast.config.GroupConfig;
+import com.hazelcast.config.MapStoreConfig;
+import com.hazelcast.core.HazelcastInstance;
+import com.hazelcast.core.IMap;
+import com.hazelcast.core.MapLoader;
+import com.hazelcast.instance.GroupProperties;
+import com.hazelcast.test.HazelcastParallelClassRunner;
+import com.hazelcast.test.HazelcastTestSupport;
+import com.hazelcast.test.TestHazelcastInstanceFactory;
+import com.hazelcast.test.annotation.QuickTest;
+
+import java.util.concurrent.Future;
+
+import static com.hazelcast.config.MapStoreConfig.InitialLoadMode.LAZY;
+import static com.hazelcast.test.TimeConstants.MINUTE;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+import org.junit.runner.RunWith;
+
+@RunWith(HazelcastParallelClassRunner.class)
+@Category(QuickTest.class)
+public class MapLoaderFailoverTest extends HazelcastTestSupport {
+
+    private static final int MAP_STORE_ENTRY_COUNT = 10000;
+    private static final int BATCH_SIZE = 100;
+    private static final int NODE_COUNT = 3;
+
+    private TestHazelcastInstanceFactory nodeFactory;
+    private CountingMapLoader mapLoader;
+
+    @Before
+    public void setUp() throws Exception {
+        nodeFactory = createHazelcastInstanceFactory(NODE_COUNT + 2);
+        mapLoader = new CountingMapLoader(MAP_STORE_ENTRY_COUNT);
+    }
+
+    @Test(timeout = MINUTE)
+    public void testDoesntLoadAgain_whenLoaderNodeGoesDown() throws Exception {
+        Config cfg = newConfig("default", LAZY);
+        HazelcastInstance[] nodes = nodeFactory.newInstances(cfg, 3);
+        HazelcastInstance hz3 = nodes[2];
+
+        String mapName = generateKeyOwnedBy(hz3);
+        IMap<Object, Object> map = nodes[0].getMap(mapName);
+
+        map.size();
+        assertSizeAndLoadCount(map);
+
+        System.err.println("terminate");
```

Comment:
Also these `System.err`s use same object to synchronize on and decreases parallelism of this test suite.